src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+package mate.academy.dao.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String INSERT_REQUEST = "INSERT INTO books(title, price) VALUES(?, ?)";
+    private static final String REQUEST_BOOK_FROM_DB = "SELECT * FROM books WHERE id = ?";
+    private static final String GET_ALL_REQUEST_BOOK = "SELECT * FROM books";
+    private static final String UPDATE_REQUEST = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_REQUEST = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement =
+                     connection.prepareStatement(INSERT_REQUEST, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert book data to DB. Book:" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement getStatement = connection.prepareStatement(REQUEST_BOOK_FROM_DB);
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            while (resultSet.next()) {
+                book = getBookFromResultSet(resultSet);
+            }
```

Comment:
not resolved

Reply:
Don't agree with you in this case. I think I need this while loop for iterating data in resultSet.

Reply:
Why you need to iterate if you get one value from result set? 
"SELECT * FROM books WHERE id = ?" returns one row
you don`t need to iterate on something

Reply:
Good point. The book's id is unique. When you try to find it by id you can get only one row
Of course, `while` works, in this case, the same as `if`.
but from the spectator's perspective, it could trick.